go-controller/pkg/ovn/address_set/address_set_cleanup.go

```diff
@@ -0,0 +1,48 @@
+package addressset
+
+import (
+	"k8s.io/klog/v2"
+)
+
+// NonDualStackAddressSetCleanup cleans addresses in old non dual stack format.
+// Assumes that for every address set <name>, if there exists an address set
+// of <name_[v4|v6]>, address set <name> is no longer used and removes it.
+// This method should only be called after ensuring address sets in old format
+// are no longer being referenced from any other object.
+func NonDualStackAddressSetCleanup() error {
+	// For each address set, track if it is in old non dual stack
+	// format and in new dual stack format
+	addressSets := map[string][2]bool{}
+	err := forEachAddressSet(func(name string) {
+		shortName := truncateSuffixFromAddressSet(name)
+		info, found := addressSets[shortName]
+		if !found {
+			info = [2]bool{false, false}
+		}
+		if shortName == name {
+			// This address set is in old non dual stack format
+			info[0] = true
+		} else {
+			// This address set is in new dual stack format
+			info[1] = true
+		}
+		addressSets[shortName] = info
+	})
+	if err != nil {
+		return err
+	}
+
+	for name, info := range addressSets {
+		// If we have an address set in both old and new formats,
```

Comment:
>  Also I think we _should_ fail since if there is something external poking around In OVN
> ...
> However we also need to do something for the case "the old AS is found and the new one isn't" because there could always be some strange race or load that may result in a delay when creating the new AS, and currently we would pass up on deleting the old AS here even though we know its an old AS and know it shouldn't be there

How about logging a warning, given that those address sets should be doing no harm other than just being there? Looking at this from the safe side, best effort point of view. I can't really think of a better option.

I do have a more elaborate idea if we want to be 100% on the safe side. Might be an overkill but putting it out here:

- On `syncNamespaces/syncPolicies`, we set an external id on old address sets that correspond to existing policies/namespaces as a mark for removal.
- On this cleanup pass we just remove address sets with that external id.

Reply:
I think that's a good compromise for now... Especially considering all of the netpol code is due for a rewrite anyways 

> I do have a more elaborate idea if we want to be 100% on the safe side. Might be an overkill but putting it out here:

>On syncNamespaces/syncPolicies, we set an external id on old address sets that correspond to existing policies/namespaces as a mark for removal.
On this cleanup pass we just remove address sets with that external id.

That would work too but results in some extra nbctl calls lets just go with the log for now, especially considering that fewer and fewer people should be running into this as we mover further away from the naming conversion